db/views/unified_restrictions_v04.sql

```diff
@@ -0,0 +1,48 @@
+WITH cards_cross_restrictions_and_snapshots AS (
+    SELECT
+        cards.id as card_id,
+        restrictions.id as restriction_id,
+        snapshots.id as snapshot_id,
+        snapshots.format_id as format_id,
+        snapshots.card_pool_id as card_pool_id,
+        snapshots.date_start as snapshot_date_start
+    FROM
+        cards, restrictions JOIN snapshots ON restrictions.id = snapshots.restriction_id
+)
+SELECT
+    cards_cross_restrictions_and_snapshots.format_id,
+    cards_cross_restrictions_and_snapshots.card_pool_id,
+    cards_cross_restrictions_and_snapshots.snapshot_id,
+    cards_cross_restrictions_and_snapshots.snapshot_date_start,
+    cards_cross_restrictions_and_snapshots.restriction_id,
+    cards_cross_restrictions_and_snapshots.card_id,
+    restrictions_cards_banned.restriction_id IS NOT NULL
+        OR restrictions_cards_restricted.restriction_id IS NOT NULL
+        OR restrictions_cards_points.restriction_id IS NOT NULL
+        OR restrictions_cards_global_penalty.restriction_id IS NOT NULL
+        OR restrictions_cards_universal_faction_cost.restriction_id IS NOT NULL
+        as in_restriction,
+    CASE WHEN restrictions_cards_banned.restriction_id IS NOT NULL THEN true ELSE false END AS is_banned,
+    CASE WHEN restrictions_cards_restricted.restriction_id IS NOT NULL THEN true ELSE false END AS is_restricted,
```

Comment:
if you want to get the same outcome, but keep this view minimally changed, you could change:

```
    COALESCE(restrictions_cards_global_penalty.value, 0) AS global_penalty,
```
to
```
    CASE WHEN restrictions_cards_global_penalty.restriction_id IS NOT NULL THEN 1 ELSE 0 END as global_penalty,
```

Reply:
oh, sorry, i misread from the db.schema diff.

This is all good here.